File: Java_Programs/test/com/example/tests/Test_Program_7.java
package com.example.tests;

import com.example.programs.Program_7;
import com.example.utils.TestUtils;
import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;

public class Test_Program_7 {
    @Test
    public void testFindCommonElements() {
        // Define test cases
        Object[][] testCases = {
                {1, new int[]{1, 2, 3}, new int[]{2, 3, 4}, new int[]{3, 4, 5}, new int[]{3}},
                {2, new int[]{10, 20, 30}, new int[]{20, 25, 30}, new int[]{20, 30, 40}, new int[]{20,30}},
                {3, new int[]{1, 3, 5}, new int[]{2, 4, 6}, new int[]{7, 8, 9}, new int[]{}}
        };

        for (Object[] testCase : testCases) {
            int num = (int) testCase[0];
            int[] arr1 = (int[]) testCase[1];
            int[] arr2 = (int[]) testCase[2];
            int[] arr3 = (int[]) testCase[3];
            int[] expectedOutput = (int[]) testCase[4];
            int[] actualOutput = Program_7.findCommonElements(arr1,arr2, arr3);

            // Print test case results
            TestUtils.printTestCase(num, new int[][]{arr1,arr2, arr3}, expectedOutput, actualOutput);

            // Assert the result
            assertArrayEquals(expectedOutput, actualOutput);
        }
    }
}

File: Python_Programs/test/test_Program_7.py
import unittest
from src.Program_7 import find_common_elements
from utils.test_utils import print_test_case


class TestProgram7(unittest.TestCase):

    def test_max_path_sum(self):
        test_cases = [
            (1, [1, 2, 3], [2, 3, 4], [3, 4, 5], [3]),
            (2, [10, 20, 30], [20, 25, 30], [20, 30, 40], [20, 30]),
            (3, [1, 3, 5], [2, 4, 6], [7, 8, 9], [])
        ]

        for num, arr1, arr2, arr3, expected_output in test_cases:
            actual_output = find_common_elements(arr1, arr2, arr3)
            print_test_case(num, [arr1, arr2, arr3], expected_output, actual_output)
            self.assertEqual(actual_output, expected_output)


if __name__ == "__main__":
    unittest.main()
